matrix/rotate_matrix.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+
+"""
+	In this problem, we want to rotate the matrix elements by 90, 180, 270 (counterclockwise)
+	Discussion in stackoverflow: 
+   https://stackoverflow.com/questions/42519/how-do-you-rotate-a-two-dimensional-array
+"""
+
+
+def rotate_90(matrix: [[]]):
+    """
+    rotate_90([[1, 2, 3, 4], [5, 6, 7, 8], [9, 10, 11, 12], [13, 14, 15, 16]])
+    [[4, 8, 12, 16], [3, 7, 11, 15], [2, 6, 10, 14], [1, 5, 9, 13]]
+    """
+    
+    transpose(matrix)
+    reverse_row(matrix)
```

Comment:
add the line __return matrix__ to the end of each function and put the doctest back in place.  This give you doctests and it enables the functionality __transpose(reverse_row(matrix))__.